upi/azure/02_storage.json

```diff
@@ -1,41 +1,159 @@
 {
-  "$schema" : "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
-  "contentVersion" : "1.0.0.0",
-  "parameters" : {
-    "baseName" : {
-      "type" : "string",
-      "minLength" : 1,
-      "metadata" : {
-        "description" : "Base name to be used in resource names (usually the cluster's Infra ID)"
+  "$schema": "https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#",
+  "contentVersion": "1.0.0.0",
+  "parameters": {
+    "architecture": {
+      "type": "string",
+      "metadata": {
+        "description": "The architecture of the Virtual Machines"
+      },
+      "defaultValue": "x64",
+      "allowedValues": [
+        "Arm64",
+        "x64"
+      ]
+    },
+    "baseName": {
+      "type": "string",
+      "minLength": 1,
+      "metadata": {
+        "description": "Base name to be used in resource names (usually the cluster's Infra ID)"
+      }
+    },
+    "imageRelease": {
+      "type": "string",
+      "metadata": {
+        "description": "RHCOS release image version - used when creating the image definition in the gallery"
       }
     },
-    "vhdBlobURL" : {
-      "type" : "string",
-      "metadata" : {
-        "description" : "URL pointing to the blob where the VHD to be used to create master and worker machines is located"
+    "storageAccountId": {
+      "type": "string",
+      "metadata": {
+        "description": "The ID of the Storage Account"
```

Comment:
whether can use storage account name, instead of the id ?

Reply:
It just so happens we can! I have updated the code.